File: resize_images.py
import os
from PIL import Image

def process_image(src, dest, size):
    try:
        with Image.open(src) as img:
            img = img.resize(size, Image.Resampling.LANCZOS)
            img.save(dest, 'WEBP')
            print(f"Processed {src} -> {dest} (Size: {size})")
    except Exception as e:
        print(f"Error processing {src}: {e}")

public_dir = r"c:\Users\User\Documents\A Hotelly\hotelly-landing-page\public"

process_image(os.path.join(public_dir, "hotelly-concierge.jpg"), os.path.join(public_dir, "hotelly-concierge.webp"), (662, 662))
process_image(os.path.join(public_dir, "icon.png"), os.path.join(public_dir, "icon.webp"), (105, 105))
process_image(os.path.join(public_dir, "hotelly.png"), os.path.join(public_dir, "hotelly.webp"), (219, 70))

File: resize_images.js
const sharp = require('sharp');
const path = require('path');

async function processImage(filename, width, height) {
  const input = path.join(__dirname, 'public', filename);
  const ext = path.extname(filename);
  const output = path.join(__dirname, 'public', path.basename(filename, ext) + '.webp');
  
  try {
    await sharp(input)
      .resize(width, height)
      .webp()
      .toFile(output);
    console.log(`Processed ${filename} to ${output} (${width}x${height})`);
  } catch (err) {
    console.error(`Error processing ${filename}:`, err);
  }
}

async function main() {
  await processImage('hotelly-concierge.jpg', 662, 662);
  await processImage('icon.png', 105, 105);
  await processImage('hotelly.png', 219, 70);
}

main();
